language/move-prover/tests/sources/verify_vector.move

```diff
@@ -89,13 +89,13 @@ module VerifyVector {
         let len = Vector::length(v);
         if (len == 0) return ();
 
-        let front_index = 0;
-        let back_index = len -1;
-        while (front_index < back_index) {
-            Vector::swap(v, front_index, back_index);
-            front_index = front_index + 1;
-            back_index = back_index - 1;
-        }
+//        let front_index = 0;
```

Comment:
Was this failing before? If you want to keep this, please add a TODO or FIXME so we don't get a surprise.

Reply:
It was not failing, but my assessment is that we don't accurately model loops in the prover.  If we do want to support them, I would advocate for adding basic loop invariants for the types - then we could re-enable this code.  By the way, we will need such assertions anyway to avoid spurious counterexamples.

Reply:
We are in the process of exploring loop invariants. I suggest we keep this around not to forget that we want to solve this. 

There is a better thing we could do. We now have a flag in the prover which lets it only generate the _verify functions if there is an associated spec block. So you want see any issues with functions with loops wo/ specs until they are actually called from functions with specs. This flag is by default off, but we could make in tests or in general on.

@junkil-park 
@DavidLDill